### ISSUE TITLE
possible Logic inversion in latest update

### ISSUE THREAD

**Issue by FergalMohan** [rs/SDWebImage#1662]
Hi guys, not sure if this is something that comes from incorrect use of SDWebImage on my end but the upgrade from 3.8.1 to 3.8.2 ended up being breaking changes for me. Upon further analysis the line 461 (below) in SDWebImageDownloaderOperation.m seems to be causing the grief. Any chance it was mistakenly changed to NSURLSessionAuthChallengeCancelAuthenticationChallenge from NSURLSessionAuthChallengePerformDefaultHandling ? 
Fergal
<<
                disposition = NSURLSessionAuthChallengeCancelAuthenticationChallenge;
>>
                disposition = NSURLSessionAuthChallengePerformDefaultHandling;
<<

### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / tvos
 Platform Version        | e.g. 8.0
 SDWebImage Version      | e.g. 3.7.6
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 7.3
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by FergalMohan** [comment]
Never mind it looks like it was an undocumented change by one of my colleagues. Sorry about the confusion.
